### PR TITLE
[DOCS] Adds note about disabled Endpoint features in K8s

### LINKED ISSUE #2801: Document disabled Endpoint features when running within Kubernetes.
## Description 

When Elastic Endpoint is deployed within Kubernetes, some Endpoint functionality is disabled. We should list them.

The are two disabled features: Malware Protection and Host Isolation.

## Acceptance Test Criteria

As a user, I would like to be informed that certain features are disabled when deploying within kubernetes while viewing deployment documentation (with the assumption that the user may not know to ask).

### PR DESCRIPTION
**Pull request by benironside**
Fixes #2801 by adding a note that mentions that when Elastic Endpoint is running in Kubernetes, the Malware Prevention and Host Isolation features are disabled.

Preview: [Kubernetes dashboard ](https://security-docs_2890.docs-preview.app.elstc.co/guide/en/security/master/cloud-nat-sec-kubernetes-dashboard.html)

### CODE REVIEW

File: docs/cloud-native-security/cloud-nat-sec-kubernetes-dashboard.asciidoc [hunk 1]
```diff
@@ -53,6 +53,8 @@ To collect session data for the dashboard, you'll deploy a Kubernetes DaemonSet
 |**GKE**| Regular (default channel): 1.21 and 1.22; Stable: 1.20 and 1.21; Rapid: 1.22 and 1.23 | Container-optimized OS (COS), Ubuntu
 |=====================
 
+NOTE: When running within Kubernetes, {elastic-endpoint}'s <<host-isolation-ov,Host Isolation>> and <<malware-proection,Malware Protection>> features are disabled.
```
Comment (nastasha-solomon) [inline docs/cloud-native-security/cloud-nat-sec-kubernetes-dashboard.asciidoc:56]: I'm fairly certain that neither features need to be capitalized when they're generally mentioned in the docs. @joepeeples thoughts?

Comment (nastasha-solomon) [reply docs/cloud-native-security/cloud-nat-sec-kubernetes-dashboard.asciidoc:56]:
```suggestion
NOTE: When running within Kubernetes, {elastic-endpoint}'s <<host-isolation-ov,host isolation>> and <<malware-proection,malware protection>> features are disabled.
```

File: docs/dashboards/kubernetes-dashboard.asciidoc [hunk 2]
```diff
@@ -54,6 +54,8 @@ To collect session data for the dashboard, you'll deploy a Kubernetes DaemonSet
 |**GKE**| Regular (default channel): 1.21 and 1.22; Stable: 1.20 and 1.21; Rapid: 1.22 and 1.23 | Container-optimized OS (COS), Ubuntu
 |=====================
 
+NOTE: When running within Kubernetes, {elastic-endpoint}'s <<host-isolation-ov,Host Isolation>> and <<malware-proection,Malware Protection>> features are disabled.
```
Comment (nastasha-solomon) [inline docs/dashboards/kubernetes-dashboard.asciidoc:57]:
```suggestion
NOTE: When running within Kubernetes, {elastic-endpoint}'s <<host-isolation-ov,host isolation>> and <<malware-proection,malware protection>> features are disabled.
```

**Review by nastasha-solomon — COMMENTED** [review]
Just left one suggestion - thanks!

File: docs/dashboards/kubernetes-dashboard.asciidoc (hunk 2 above)
Comment (benironside) [reply docs/dashboards/kubernetes-dashboard.asciidoc:57]: looks good to me! Thanks

File: docs/cloud-native-security/cloud-nat-sec-kubernetes-dashboard.asciidoc (hunk 1 above)
Comment (joepeeples) [reply docs/cloud-native-security/cloud-nat-sec-kubernetes-dashboard.asciidoc]: +1 to lowercase, which has already been implemented 👍

**Review by joepeeples — APPROVED** [review]
LGTM!

**Review by nicholasberlin — APPROVED** [review]

**Pull request merged**